src/sqs/error-handlers.ts

```diff
@@ -85,10 +86,16 @@ const maybeHandleNonRetryableResponseCode = <MessagePayload extends BaseMessageP
 
 const maybeHandleRateLimitingError = <MessagePayload extends BaseMessagePayload>(error: Error, context: SQSMessageContext<MessagePayload>): ErrorHandlingResult | undefined => {
 	if (error instanceof GithubClientRateLimitingError) {
+		const DELAY_BUFFER_STEP = 10;
 		context.log.warn({ error }, `Rate limiting error, retrying`);
-		const delaySec = error.rateLimitReset + RATE_LIMITING_DELAY_BUFFER_SEC - (Date.now() / 1000);
-		return { retryable: true, retryDelaySec: delaySec, isFailure: true };
+		const buffer = Math.max(DELAY_BUFFER_STEP, BASE_RATE_LIMITING_DELAY_BUFFER_SEC - context.receiveCount * DELAY_BUFFER_STEP);
```

Comment:
Sill would like to see a comment clarifying this line, too ;)

Reply:
Clarified! please let me know if it makes sense though!